### PR TITLE
Creates query to recover failed recurring donations given a period and improves language for monthly report

### PR DESCRIPTION
**Pull request by clarissalimab**
Missing part from #120.

### CODE REVIEW

File: lib/query/query_recurring_donations.rb
Comment (clarissalimab) [inline lib/query/query_recurring_donations.rb]: I had to do it because I couldn't find a way to properly get the root_url from there or from FailedRecurringDonationsReport :/

Comment (wwahammy) [reply lib/query/query_recurring_donations.rb]: That's fine. There's probably better ways but it's not worth looking into it right now.

File: lib/query/query_recurring_donations.rb
Comment (wwahammy) [inline lib/query/query_recurring_donations.rb]: `:started_at` and `:end_date` are not the greatest hash names but I don't think you created them, did you?

File: lib/query/query_recurring_donations.rb
Comment (wwahammy) [inline lib/query/query_recurring_donations.rb]: If the goal here is to get a boolean, I would use `!!query[:include_last_failed_charge]` to the right side.

**Review by wwahammy — CHANGES_REQUESTED** [review]
Seems pretty good! There is one tweak but otherwise I think it should be merged.

File: lib/query/query_recurring_donations.rb
Comment (clarissalimab) [reply lib/query/query_recurring_donations.rb]: Yes! I created them. Those are to mark the beginning of the period and the ending of the period for failed recurring donations. I think I saw it like that somewhere else and reused it, but I think we can change that to something else, it would be better. What about "from_date" and "to_date"? Or "after_date" and "before_date"?

File: lib/query/query_recurring_donations.rb
Comment (wwahammy) [inline lib/query/query_recurring_donations.rb]: Question: is there a situation in where we'd want to look for failed charges during a time period but not also use the same time period for the general filtering?

Comment (wwahammy) [reply lib/query/query_recurring_donations.rb]: I *THINK* they'd always be the same but I may be overlooking something.

**Review by wwahammy — COMMENTED** [review]
Good work as always but I have some questions before I can approve or request changes!

File: lib/query/query_recurring_donations.rb
Comment (clarissalimab) [reply lib/query/query_recurring_donations.rb]: I don't think so, but I didn't follow what you think should be different.

Comment (wwahammy) [reply lib/query/query_recurring_donations.rb]: Oh, I get what's going on. There's not already a search filter for the recurring donations based upon date. I was thinking of the payments in queries.

Nevermind! 😀

File: lib/query/query_recurring_donations.rb
Comment (wwahammy) [reply lib/query/query_recurring_donations.rb]: So I'd like a way to indicate that the value for `:started_at` is inclusive of the time passed but `:end_date` is exclusive of the time passed. So what kind of names would we use to do that?

Comment (clarissalimab) [reply lib/query/query_recurring_donations.rb]: Hmm, so how do `:from_date` and `:before_date` sound to you?

Comment (wwahammy) [reply lib/query/query_recurring_donations.rb]: I like it!

**Comment by clarissalimab** [comment]
@wwahammy I included the needed changes to be able to send the same report to multiple users, can you take a look?

**Comment by wwahammy** [comment]
@clarissalimab where are we on this?

**Comment by clarissalimab** [comment]
@wwahammy it's waiting for your review again after I added the ability to have multiple users for the same report!

File: db/migrate/20211101221537_create_periodic_reports_users.rb
Comment (wwahammy) [inline db/migrate/20211101221537_create_periodic_reports_users.rb]: This should be `t.references`

**Review by wwahammy — APPROVED** [review]
Quick change requested on the migration but otherwise, if the tests pass, it should be merged.

**Review by wwahammy — APPROVED** [review]
:+1:

**Pull request merged**
